### PR TITLE
[SR-4630] Cancel BitmapValue's memory sharing

### PR DESCRIPTION
**Pull request by Pslydhh**
`select A, bitmap_union_count(B) as 'D' from table0 group by cube (C);`
- If need modify data, input Bitmap value should not be shared, it should be copied.
- Optimization may be needed in the future to reduce replication.

### CODE REVIEW

**Review by liuyehcf — DISMISSED** [review]

File: be/src/util/bitmap_value.h [hunk 1]
```diff
@@ -1375,24 +1376,23 @@ class BitmapValue {
...
                 _type = BITMAP;
                 break;
             case BITMAP: {
-                BitmapValue lhs_bitmap(*this);
-                *_bitmap -= *rhs._bitmap;
-                *rhs._bitmap -= *lhs_bitmap._bitmap;
-
+                BitmapValue intersect_bitmap(*this);
                 *_bitmap |= *rhs._bitmap;
+                *intersect_bitmap._bitmap &= *rhs._bitmap;
```
Comment (dirtysalt) [inline be/src/util/bitmap_value.h:1394]: it that logically correct?

Comment (Pslydhh) [reply be/src/util/bitmap_value.h:1394]: a⊕b = (¬a ∧ b) ∨ (a ∧¬b) = (a V b) ∧ ¬(a ∧ b)
Before modification: (¬a ∧ b) ∨ (a ∧¬b)
after modification: (a V b) ∧ ¬(a ∧ b)

Comment (Pslydhh) [reply be/src/util/bitmap_value.h:1394]: > it that logically correct?

It's the same in terms of set operations

Comment (dirtysalt) [reply be/src/util/bitmap_value.h:1394]: a more efficient & straightforward way is:

```
*_bitmap -= *rhs._bitmap;
new X(*rhs._bitmap);
X -= *_bitmap;
*_bitmap |= X;

```

Comment (dirtysalt) [reply be/src/util/bitmap_value.h:1394]: No, my way is wrong.

**Review by dirtysalt — DISMISSED** [review]

**Review by stdpain — APPROVED** [review]

**Comment by kangkaisen** [comment]
run starrocks_be_unittest

**Review by dirtysalt — APPROVED** [review]

**Pull request merged**